XenAdmin/TabPages/GeneralTabPage.cs

```diff
@@ -1359,6 +1334,133 @@ private void generateGeneralBox()
             s.AddEntry(FriendlyName("host.uuid"), GetUUID(xenObject));
         }
 
+        private static void GenerateVirtualisationStatusForGeneralBox(PDSection s, VM vm)
+        {
+            if (vm != null && vm.Connection != null && vm.power_state == vm_power_state.Running)
+            {
+                //For Dundee or higher Windows VMs
+                if (Helpers.DundeeOrGreater(vm.Connection) && vm.IsWindows)
+                {
+                    var gm = vm.Connection.Resolve(vm.guest_metrics);
+
+                    bool isIoOptimized = gm != null && gm.network_paths_optimized && gm.storage_paths_optimized;
+                    bool isReceivingUpdates = vm.auto_update_drivers;
+
+                    bool isXenPrepInProgress = false; //TODO in CP-13247  when XenPrep functions will be added
+                    bool canTurnOnAutoUpdates = false && !isReceivingUpdates && !isXenPrepInProgress; //TODO in CP-13247: remove &&false when XenPrep functions have been added
+
+                    bool isManagementAgentInstalled = vm.HasRDP; //HasRDP is the way to detect .Net/Management Agent
+
+                    bool canInstallIoDriversAndManagementAgent = InstallToolsCommand.CanExecute(vm) && !isIoOptimized && !isManagementAgentInstalled;
+                    bool canInstallManagementAgentOnly = InstallToolsCommand.CanExecute(vm) && !canInstallIoDriversAndManagementAgent && !isManagementAgentInstalled;

```

Comment:
I think this would be clearer with isIoOptimized instead of !canInstallIoDriversAndManagementAgent on the RHS of this line.